server/firebase.js

```diff
@@ -1,35 +1,49 @@
+/**
+ * Common firebase functions
+ * @exports getDownloadUrl  entry image URL from firebase storage
+ * @exports getToken create auth token for access to Firebase Storage
+ */
+
 const admin = require('firebase-admin');
 const { cert } = require('firebase-admin/app');
 const { getAuth } = require('firebase-admin/auth');
 const { getStorage } = require('firebase-admin/storage');
 
+const { IS_FIREBASE_ON } = require('./env');
+
 const { FIREBASE_CLIENT_EMAIL, FIREBASE_PRIVATE_KEY } = process.env;
 
 const LIFETIME = 60 * 24 * 60 * 60 * 1000; // 60 days -> milliseconds
 
-admin.initializeApp({
-  credential: cert({
-    clientEmail: FIREBASE_CLIENT_EMAIL,
-    privateKey: Buffer.from(FIREBASE_PRIVATE_KEY, 'base64').toString('ascii'),
-    projectId: 'vexillology-contests',
-  }),
-});
-
-const storage = getStorage();
-
-exports.getDownloadUrl = async (image) => {
-  try {
-    const [downloadUrl] = await storage
-      .bucket('vexillology-contests.appspot.com')
-      .file(`images/${image}`)
-      .getSignedUrl({ action: 'read', expires: Date.now() + LIFETIME });
-    return downloadUrl;
-  } catch (err) {
-    return null;
-  }
-};
-
-exports.getToken = async (uid) => {
-  const token = await getAuth().createCustomToken(uid);
-  return token;
-};
+if (IS_FIREBASE_ON) {
+  admin.initializeApp({
+    credential: cert({
+      clientEmail: FIREBASE_CLIENT_EMAIL,
+      privateKey: Buffer.from(FIREBASE_PRIVATE_KEY, 'base64').toString('ascii'),
+      projectId: 'vexillology-contests',
+    }),
+  });
+
+  const storage = getStorage();
+
+  const getDownloadUrl = async (image) => {
+    try {
+      const [downloadUrl] = await storage
+        .bucket('vexillology-contests.appspot.com')
+        .file(`images/${image}`)
+        .getSignedUrl({ action: 'read', expires: Date.now() + LIFETIME });
+      return downloadUrl;
+    } catch (err) {
+      return null;
+    }
+  };
+
+  const getToken = async (uid) => {
+    const token = await getAuth().createCustomToken(uid);
+    return token;
+  };
+
+  module.exports = {
+    getDownloadUrl, getToken,
+  };
```

Comment:
Are there no exports if firebase is off? Would that cause a runtime error something like `getDownloadUrl is not a function`?

Reply:
Firebase is always on in production, so we won't hit an issue there.
In my dev environment, I am only pulling from Postgres so I have not hit an error.

I'm not sure if you will run into an issue in your dev or staging env...